tests/unit/sharded_test.cc

```diff
@@ -201,3 +201,32 @@ SEASTAR_THREAD_TEST_CASE(invoke_on_all_sharded_arg) {
     srv.stop().get();
     arg.stop().get();
 }
+
+SEASTAR_THREAD_TEST_CASE(invoke_on_modifiers) {
+    class checker {
+    public:
+        future<> fn(int a) {
+            return make_ready_future<>();
+        }
+        future<> fn(int a) const {
+            return make_ready_future<>();
+        }
+    };
+
+    seastar::sharded<checker> srv;
+    srv.start().get();
+    int a = 42;
+
+    srv.invoke_on_all([a] (checker& s) { return s.fn(a); }).get();
```

Comment:
i'd suggest add a test case which pass a `const &` to the lambda, like
```c++
srv.invoke_on_all([a] (const checker& s) { return s.fn(a); }).get();
```

Reply:
It's below, and it's commented out, because it doesn't work currently: #2278

Reply:
it's not below. could you please read the comment again? please.

Reply:
OK, I missed that you suggest to have `const checker&`, but _not_ `const_cast<const ...>(srv)` part. What for?

Reply:
hmm, just for the sake of completeness. but it's not important.